traffic_portal/app/src/common/modules/dialog/select/lock/dialog.select.lock.tpl.html

```diff
@@ -0,0 +1,69 @@
+<!--
+Licensed to the Apache Software Foundation (ASF) under one
+or more contributor license agreements.  See the NOTICE file
+distributed with this work for additional information
+regarding copyright ownership.  The ASF licenses this file
+to you under the Apache License, Version 2.0 (the
+"License"); you may not use this file except in compliance
+with the License.  You may obtain a copy of the License at
+  http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+<div class="modal-header">
+    <button type="button" class="close" ng-click="cancel()"><span aria-hidden="true">&times;</span><span class="sr-only">Close</span></button>
+    <h4 class="modal-title">Create CDN Lock</h4>
+</div>
+<div class="modal-body">
+    <form name="lockForm" novalidate>
+        <div class="row">
+            <div class="col-sm-12 col-md-12">
+                <div class="form-group" ng-class="{'has-error': hasError(lockForm.cdn), 'has-feedback': hasError(lockForm.cdn)}">
+                    <label class="control-label" for="cdn">CDN *
+                        <small class="input-error" ng-show="hasPropertyError(lockForm.cdn, 'required')">Required</small>
+                    </label>
+                    <select id="cdn" name="cdn" class="form-control" ng-model="lock.cdn" ng-options="cdn.name as cdn.name for cdn in cdns" required>
+                        <option value="">Select...</option>
+                    </select>
+                </div>
+            </div>
+        </div>
+        <div class="row">
+            <div class="col-sm-12 col-md-12">
+                <div class="form-group" ng-class="{'has-error': hasError(lockForm.message), 'has-feedback': hasError(lockForm.message)}">
+                    <label class="control-label" for="message">Message *
+                        <small class="input-error" ng-show="hasPropertyError(lockForm.message, 'maxlength')">Too Long</small>
+                    </label>
+                    <input id="message" name="message" type="text" class="form-control" ng-model="lock.message" ng-maxlength="256" required autofocus>
+                    <span ng-show="hasError(lockForm.message)" class="form-control-feedback"><i class="fa fa-times"></i></span>
+                </div>
+            </div>
+        </div>
+        <div class="row">
+                <div class="col-sm-12 col-md-12">
+                    <div class="form-group">
+                        <p>Type *</p>
```

Comment:
example please :)

Reply:
```html
<div class="modal-header">
    <button type="button" class="close" ng-click="cancel()"><span aria-hidden="true">&times;</span><span class="sr-only">Close</span></button>
    <h4 class="modal-title">Create CDN Lock</h4>
</div>
<div class="modal-body">
    <form name="lockForm" novalidate>
        <div class="row">
            <div class="col-sm-12 col-md-12">
                <div class="form-group" ng-class="{'has-error': hasError(lockForm.cdn), 'has-feedback': hasError(lockForm.cdn)}">
                    <label class="control-label" for="cdn">CDN *
                        <small class="input-error" ng-show="hasPropertyError(lockForm.cdn, 'required')">Required</small>
                    </label>
                    <select id="cdn" name="cdn" class="form-control" ng-model="lock.cdn" ng-options="cdn.name as cdn.name for cdn in cdns" required>
                        <option disabled hidden selected value="">Select...</option>
                    </select>
                </div>
            </div>
        </div>
        <div class="row">
            <div class="col-sm-12 col-md-12">
                <div class="form-group" ng-class="{'has-error': hasError(lockForm.message), 'has-feedback': hasError(lockForm.message)}">
                    <label class="control-label" for="message">Message *
                        <small class="input-error" ng-show="hasPropertyError(lockForm.message, 'maxlength')">Too Long</small>
                    </label>
                    <input id="message" name="message" type="text" class="form-control" ng-model="lock.message" maxlength="256" required autofocus>
                    <span ng-show="hasError(lockForm.message)" class="form-control-feedback"><i class="fa fa-times"></i></span>
                </div>
            </div>
        </div>
        <div class="row">
            <div class="col-sm-12 col-md-12">
                <fieldset class="form-group" ng-class="{'has-error': hasError(lockform.lockType)}">
                    <legend>Type</legend>
                    <input id="soft" name="lockType" type="radio" ng-model="lock.soft" ng-value="true" required>
                    <label class="has-tooltip" for="soft">Soft<div class="helptooltip">
                        <div class="helptext">A soft lock will prevent others from queueing server updates or snapshotting the locked CDN.</div>
                    </div>
                    </label><br>
                    <input id="hard" name="lockType" type="radio" ng-model="lock.soft" ng-value="false" required>
                    <label class="has-tooltip" for="soft">Hard<div class="helptooltip">
                        <div class="helptext">A hard lock will prevent others from queueing server updates or snapshotting the locked CDN. It will also block others from making changes that would change the state of the locked CDN.</div>
                    </div>
                    </label>
                </fieldset>
            </div>
        </div>
    </form>
</div>
<div class="modal-footer">
    <button type="button" class="btn action-btn" ng-click="cancel()">Cancel</button>
    <button type="button" class="btn btn-link" ng-disabled="lockForm.$invalid" ng-click="select()">Submit</button>
</div>
```

Reply:
thanks. that looks much better and i updated the tooltip a bit to look like this:

![image](https://user-images.githubusercontent.com/251272/121944379-d9f75280-cd0f-11eb-9e46-7b39c8462686.png)

but it could use some tooltip style love i think

Reply:
yeah, I think ideally that would be off to the side or underneath. The modal is too close to the top, seems like certain font settings might cut it off. Also, seems like that's making a definition list, so `<dl>`/`<dt>`/`<dd>` might be more appropriate than `<ul>`/`<li>`, though I tried that out and that **did** get cut off at the top.